pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -172,10 +174,11 @@ async def serve_requests(self, num_requests=None):
                 await self._on_serving_error(exc)
 
             finally:
-                if self._is_drain_needed:
+                if self._is_drain_needed and not self._discard_events:
```

Comment:
We can remove it from here because we already ignore in the drain signal handler.
```suggestion
                if self._is_drain_needed:
```